content/en/BestPractices/SecurityMisconfiguration.md

```diff
@@ -0,0 +1,115 @@
+
+
+A very common scenario of security misconfiguration is a verbose error message thrown by the application when it encounters unexpected inputs. These errors may convey a lot about the backend such as the framework or technology in use, the internal path of the application, code snippets, or sometimes even secret keys. The exposed information is valuable for attackers. 
+
+Another example is directory listing enabled in the webserver setting, which allows users to access the full directory structure and access files in the directories. This allows easy access to files and directories to any visitor. This could lead loss of sensitive information.  
+
+These misconfigurations can be taken care of early in the development lifecycle by software security design review and Threat modeling. Attending these misconfigurations and fixing them is simple as compared to other complex vulnerabilities in applications. There are precautions that developers can take before moving the code to production to avoid these mistakes. 
+
+Let us look into these common misconfigurations and understand how developers can prevent these issues.
+
+## Debug Enabled and verbose errors ##
+
+One of the common misconfigurations is developers leaving the debug enabled in the prod environment. The debug allows exposure of more information than expected. Debug info may contain sensitive data such as application internal paths, keys or secrets, or backend code.  
+
+Depending on the framework or development language in use, it is required to configure secure settings and disable debug settings. For example, in .NET applications by configuring Web.config file properly debug can be disabled. 
+
+In .NET applications modify the Web.config file
+
+```
+<compilation debug="false"/>
+```
+Or modify the Machine.config file
+
+````
+<compilation debug="false"/>
+````
+
+Similarly, applications if not configured with custom errors may leak verbose errors or stack trace which exposes sensitive information. The application should be configured with custom errors and any stack trace should be prevented to be exposed to users. 
+
+A simple example to configure custom errors in .NET application in Web.config file:
+
+````
+<system.web>  
+<customErrors mode="RemoteOnly" defaultRedirect="Error.aspx">  
+<error statusCode="401" redirect="CustomPage.aspx" /> 
+<error statusCode="403" redirect="CustomPage.aspx" /> 
+<error statusCode="404" redirect="NotFound.aspx" /> 
+<error statusCode="500" redirect="Error.aspx" />  
+</customErrors>  
+</system.web>  
+
+ ````
+
+## Directory listing ##
+
+Directory listing a very common misconfiguration at the webserver exposes all the directories and files to the users.  A simple google search with ` intitle: "index of /"` list many misconfigured websites which have directory listing enabled. This misconfiguration may leak protected files and grant unauthorized access to the application directories and files. 
```

Comment:
I usually hesitate to describe vulnerabilities as "common" or "uncommon", because it can be a little subjective. If I've never heard of it, but you've dealt with five of them in the last hour, we have slightly different ideas of what's subjective.